### PR TITLE
Adds the gravity generator

### PR DESCRIPTION
**Pull request by printer16**
Adds the gravity generator and maps it in here:
gravity
Works like it did in old code with some minor additions. The sprite is the one TG uses.

### CODE REVIEW

**Comment by SoundScopes** [comment]
This will need rebasing to development branch

**Comment by SoundScopes** [comment]
Oh also if this works just like the old one then you might want to make it multi-z compatible. As it stands it will only make gravity on the z-level it is on.

**Comment by printer16** [comment]
Rebased. Also works during testing as  is, but updated to include the other Z's in the list.

**Comment by printer16** [comment]
Rebased. When tested it works as is removing/providing gravity from station levels.

File: code/modules/events/gravity.dm [hunk 1]
```diff
@@ -6,18 +6,18 @@
 	endWhen = rand(15, 60)
 
 /datum/event/gravity/announce()
-	command_announcement.Announce("Feedback surge detected in mass-distributions systems. Artificial gravity has been disabled whilst the system reinitializes. Further failures may result in a gravitational collapse and formation of blackholes.", "Gravity Failure")
+	command_announcement.Announce("Feedback surge detected in Gravity Generation systems. Artificial gravity has been disabled whilst the system reinitializes. Further failures may result in a gravitational collapse and formation of blackholes.", "Gravity Failure")
```
Comment (Lohikar) [inline code/modules/events/gravity.dm]: Minor nitpick: This should just be `gravity generation` (lowercase).

File: code/modules/events/gravity.dm [hunk 2]
```diff
@@ -6,18 +6,18 @@
...
+	for(var/obj/machinery/gravity_generator/main/A in gravity_generators)
+		A.eventshutofftoggle()
 
 /datum/event/gravity/end()
 	if(!gravity_is_on)
 		gravity_is_on = 1
 
-		for(var/area/A in the_station_areas)
-			A.gravitychange(gravity_is_on, A)
+		for(var/obj/machinery/gravity_generator/main/A in gravity_generators)
```
Comment (Lohikar) [inline code/modules/events/gravity.dm]: Type-less loop.

File: code/modules/mob/living/carbon/human/human_movement.dm [hunk 3]
```diff
@@ -124,3 +124,13 @@
 			footstep++
 			if (footstep % 2)
 				playsound(src, T.footstep_sound, 40, 1)
+
+/mob/living/carbon/human/mob_has_gravity()
+	. = ..()
+	if(!.)
```
Comment (Lohikar) [inline code/modules/mob/living/carbon/human/human_movement.dm]: `if (!. && mob_negates_gravity())`

File: code/modules/mob/living/carbon/human/human_movement.dm [hunk 4]
```diff
@@ -124,3 +124,13 @@
 			footstep++
 			if (footstep % 2)
 				playsound(src, T.footstep_sound, 40, 1)
+
+/mob/living/carbon/human/mob_has_gravity()
+	. = ..()
+	if(!.)
+		if(mob_negates_gravity())
+			. = 1
+			return .
```
Comment (Lohikar) [inline code/modules/mob/living/carbon/human/human_movement.dm]: This return is pointless.

File: code/modules/power/gravitygenerator.dm [hunk 5]
```diff
@@ -1,140 +1,455 @@
...
-	icon = 'icons/obj/computer.dmi'
-	icon_state = "airtunnel0e"
-	anchored = 1
-	density = 1
-	var/obj/machinery/gravity_generator = null
-
-
-/obj/machinery/gravity_generator/
-	name = "Gravitational Generator"
+var/list/gravity_generators = list() // We will keep track of this by adding new gravity generators to the list, and keying it with the z level.
```
Comment (Lohikar) [inline code/modules/power/gravitygenerator.dm]: Put this list in SSmachinery please.

**Pull request merged**